### PR TITLE
chore: add test for withdrawal

### PR DESCRIPTION
**Pull request by haongo138**
**What does this PR do?**

-   [x] Test for withdrawal feature

### CODE REVIEW

File: src/commands/withdraw/index/processor.test.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        tx_url:
+          "https://ftmscan.com/tx/0x3b47c97f3f7bf3b462eba7b2b546f927a3b59be7103ff0439123123",
+      },
+    }
+    Defi.getInsuffientBalanceEmbed = jest.fn().mockResolvedValueOnce(null)
+    Defi.offchainDiscordWithdraw = jest
+      .fn()
+      .mockResolvedValueOnce(mockedResponse)
+    await processor.withdraw(msg, args)
+    expect(msg.author.send).toHaveBeenCalledTimes(1)
```
Comment (anhnh12) [inline src/commands/withdraw/index/processor.test.ts:59]: call with what arg

File: src/commands/withdraw/index/processor.test.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    const args = [
+      "withdraw",
+      "1",
+      "ftm",
+      "0xE409E073eE7474C381BFD9b3f88098499123123",
+    ]
+    Defi.getInsuffientBalanceEmbed = jest.fn().mockResolvedValueOnce({})
+    Defi.offchainDiscordWithdraw = jest.fn()
+    await processor.withdraw(msg, args)
+    expect(Defi.offchainDiscordWithdraw).not.toHaveBeenCalled()
```
Comment (anhnh12) [inline src/commands/withdraw/index/processor.test.ts:72]: assert output result

File: src/commands/withdraw/index/processor.test.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  test("invalid amount", async () => {
+    const args = [
+      "withdraw",
+      "-1",
+      "ftm",
+      "0xE409E073eE7474C381BFD9b3f88098499123123",
+    ]
+    await expect(processor.withdraw(msg, args)).rejects.toThrow(
+      DiscordWalletTransferError
```
Comment (anhnh12) [inline src/commands/withdraw/index/processor.test.ts:83]: assert error msg

File: src/commands/withdraw/index/processor.test.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      "ftm",
+      "0xE409E073eE7474C381BFD9b3f88098499123123",
+    ]
+    Defi.offchainDiscordWithdraw = jest.fn().mockResolvedValueOnce({
+      ok: false,
+      data: null,
+    })
+    await expect(
+      processor.withdrawSlash(interaction, args[1], args[2], args[3])
+    ).rejects.toThrow(APIError)
```
Comment (anhnh12) [inline src/commands/withdraw/index/processor.test.ts:130]: assert error msg

File: src/commands/withdraw/index/text.ts [hunk 5]
```diff
@@ -42,7 +42,7 @@ const run = async (msg: Message) => {
     ],
   })
 
-  if (msg.guild !== null) {
+  if (msg.guildId !== null) {
```
Comment (anhnh12) [inline src/commands/withdraw/index/text.ts:45]: revert this

File: src/commands/withdraw/index/text.test.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  let msg: Message
+  const withdrawCmd = commands["withdraw"]
+
+  beforeEach(() => (msg = mockdc.cloneMessage()))
+  afterEach(() => jest.clearAllMocks())
+
+  test("invalid amount", async () => {
+    msg.content = `$withdraw -1 ftm`
+    // const output = () as RunResult<MessageOptions>
+    await expect(withdrawCmd.run(msg)).rejects.toThrow(InternalError)
```
Comment (anhnh12) [inline src/commands/withdraw/index/text.test.ts:23]: assert output result

File: src/commands/withdraw/index/text.test.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    msg.author.avatarURL = jest.fn().mockResolvedValueOnce(null)
+    msg.reply = jest.fn().mockResolvedValueOnce(undefined)
+
+    Defi.getInsuffientBalanceEmbed = jest.fn().mockResolvedValueOnce(null)
+    jest
+      .spyOn(processor, "getDestinationAddress")
+      .mockResolvedValueOnce("0xE409E073eE7474C381BFD9b3f88098499123123")
+    jest.spyOn(processor, "withdraw").mockResolvedValueOnce(undefined)
+    await withdrawCmd.run(msg)
+    expect(processor.withdraw).toHaveBeenCalledWith(msg, args)
```
Comment (anhnh12) [inline src/commands/withdraw/index/text.test.ts:63]: assert output result

**Comment by anhnh12** [comment]
may cai assert neu dc xai `assertRunResult` luon nha e
k assert tung field rieng

**Review by anhnh12 — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Pull request merged**